packages/legend-query-builder/src/stores/QueryBuilderResultState.ts

```diff
@@ -238,6 +241,34 @@ export class QueryBuilderResultState {
     }
   }
 
+  getExecutionResultLimit = (): number =>
+    Math.min(
+      this.queryBuilderState.fetchStructureState.implementation instanceof
+        QueryBuilderTDSState &&
+        this.queryBuilderState.fetchStructureState.implementation
+          .resultSetModifierState.limit
+        ? this.queryBuilderState.fetchStructureState.implementation
+            .resultSetModifierState.limit
+        : Number.MAX_SAFE_INTEGER,
+      this.previewLimit,
+    );
+
+  getTruncatedExecutionResultIfExceed = (
```

Comment:
Don't really want to have to call this upon each render and having to do deep clone. 

Can't we all handle this in the `runQuery` function ?
maybe replace executionResult with 
`executionResultState?: ExecutionResultState`

```
ExecutionResultState
{
  result: ExecutionResult:
  isOverflowing: boolean (set to false by default)
}

```
Then when you get the result back, if it overflows you set that to true and at the time of setting the executionResult you remove the result from the orginal object (no deep clone). as you should never need that value again for anything. 
My fear is we use may orginal object for other things causing other issues and there is no reason to deep clone as that extra value is only for that warning purpose. 
Might as well handle this in run query.